libs/portal-integration-angular/src/lib/core/components/button-dialog/button-dialog.component.ts

```diff
@@ -0,0 +1,194 @@
+      return ocxDialogButtonClickedResult
+    }
+    return from(Promise.resolve(ocxDialogButtonClickedResult))
+  }
+
+  loadComponent() {
+    if (this.dynamicDialogConfig.data !== undefined) {
+      this.setUpDialogDataForDynamicConfig()
+    } else {
+      this.setUpDialogDataForInput()
+    }
+  }
+
+  setUpDialogDataForDynamicConfig() {
+    const dynamicConfigData: ButtonDialogData = this.dynamicDialogConfig.data
+    if (dynamicConfigData.config !== undefined) {
+      const dialogConfig = dynamicConfigData.config
+      if (dialogConfig.primaryButtonDetails !== undefined && dialogConfig.primaryButtonDetails.key !== undefined) {
+        this.dialogData.config.primaryButtonDetails = dialogConfig.primaryButtonDetails
+      }
+      if (dialogConfig.secondaryButtonIncluded !== undefined) {
+        this.dialogData.config.secondaryButtonIncluded = dialogConfig.secondaryButtonIncluded
+      }
+      if (dialogConfig.secondaryButtonDetails !== undefined && dialogConfig.secondaryButtonDetails.key !== undefined) {
+        this.dialogData.config.secondaryButtonDetails = dialogConfig.secondaryButtonDetails
+      }
+    }
+    if (dynamicConfigData.component !== undefined) {
+      this.dialogData.component = dynamicConfigData.component
+    }
+    if (dynamicConfigData.componentData !== undefined) {
+      this.dialogData.componentData = dynamicConfigData.componentData
+    }
+
+    const viewContainerRef = this.dialogHost
+    viewContainerRef.clear()
+
+    if (this.dialogData.component) {
+      const componentRef = viewContainerRef.createComponent<any>(this.dialogData.component)
+      //check for DialogPrimaryButtonDisabled and DialogSecondaryButtonDisabled interfaces
+      if ('primaryButtonEnabled' in componentRef.instance) {
+        this.primaryButtonEnabled = false
+        componentRef.instance.primaryButtonEnabled.subscribe({
+          next: (enabled: boolean) => {
+            this.primaryButtonEnabled = enabled
+          },
+        })
+      }
+      if ('secondaryButtonEnabled' in componentRef.instance) {
+        this.secondaryButtonEnabled = false
+        componentRef.instance.secondaryButtonEnabled.subscribe({
+          next: (enabled: boolean) => {
+            this.secondaryButtonEnabled = enabled
+          },
+        })
+      }
+      //populate container
+      Object.keys(this.dialogData.componentData).forEach((k) => {
+        componentRef.instance[k] = this.dialogData.componentData[k]
+      })
+      this.componentRef = componentRef
+    }
+  }
+
+  setUpDialogDataForInput() {
+    this.dialogData.component = undefined
+    this.dialogData.componentData = undefined
+    if (this.config !== undefined) {
+      if (this.config.primaryButtonDetails !== undefined && this.config.primaryButtonDetails.key !== undefined) {
+        this.dialogData.config.primaryButtonDetails = this.config.primaryButtonDetails
+      }
+      if (this.config.secondaryButtonIncluded !== undefined) {
```

Comment:
Using double exclamation mark does not work since for all secondaryButtonIncluded possible values:
!!undefined -> false
!!false -> false
!!true -> true

By default secondaryButtonIncluded is set to true. If false is provided the if statement does not execute